src/ui/ui_osd_element_pos.c

```diff
@@ -0,0 +1,665 @@
+#include "ui/ui_osd_element_pos.h"
+
+#include <stdio.h>
+
+#include <log/log.h>
+#include <lvgl/lvgl.h>
+#include <minIni.h>
+
+#include "core/common.hh"
+#include "core/osd.h"
+#include "driver/hardware.h"
+#include "driver/oled.h"
+#include "log/log.h"
+#include "ui/page_common.h"
+#include "ui/page_osd.h"
+#include "util/math.h"
+
+#define CANVAS_WIDTH  445
+#define CANVAS_HEIGHT 365
+
+#define OSD_ELEMENT_MIN_X_POS 0
+#define OSD_ELEMENT_MAX_X_POS 1280
+#define OSD_ELEMENT_MIN_Y_POS 0
+#define OSD_ELEMENT_MAX_Y_POS 720
+
+// used for slider scroll acceleration
+#define SLIDER_SCROLL_MIN_INTERVAL  10  // min time interval between scrolls (ms)
+#define SLIDER_SCROLL_MAX_INTERVAL  100 // max time interval between scrolls (ms)
+#define SLIDER_SCROLL_MIN_STEP_SIZE 1
+#define SLIDER_SCROLL_MAX_STEP_SIZE 10
+
+enum {
+    ROW_OSD_MODE = 0,
+    ROW_OSD_ELEMENT,
+    ROW_OSD_SHOW_ELEMENT,
+    ROW_OSD_ELEMENT_POS_X,
+    ROW_OSD_ELEMENT_POS_Y,
+    ROW_SAVE,
+    ROW_CANCEL,
+    ROW_RESET_ELEMENTS,
+
+    ROW_COUNT
+};
+
+typedef enum {
+    UI_STATE_SCROLL = 0,
+    UI_STATE_FOCUSED
+} ui_state_t;
+
+typedef enum {
+    CONFIRMATION_UNCONFIRMED = 0,
+    CONFIRMATION_CONFIRMED,
+    CONFIRMATION_TIMEOUT
+} ui_confirmation_t;
+
+// a list of all osd elements
+// must contain the same elements, in the same order, as osd_element_list[]
+typedef enum {
+    OSD_ELEMENT_TOP_FAN_SPEED = 0,
+    OSD_ELEMENT_LATENCY_LOCK,
+    OSD_ELEMENT_VTX_TEMP,
+    OSD_ELEMENT_VRX_TEMP,
+    OSD_ELEMENT_BATTERY_LOW,
+    OSD_ELEMENT_CHANNEL,
+    OSD_ELEMENT_SD_REC,
+    OSD_ELEMENT_VLQ,
+    OSD_ELEMENT_ANT0,
+    OSD_ELEMENT_ANT1,
+    OSD_ELEMENT_ANT2,
+    OSD_ELEMENT_ANT3,
+    OSD_ELEMENT_GOGGLE_TEMP_TOP,
+    OSD_ELEMENT_GOGGLE_TEMP_LEFT,
+    OSD_ELEMENT_GOGGLE_TEMP_RIGHT,
+
+    OSD_ELEMENTS_TOTAL
+} osd_elements_t;
+
+typedef struct {
+    char *name;
+    char *name_settings;
+} osd_element_t;
+
+// ###########################################################################
+// ########################## local variables start ##########################
+// ###########################################################################
```

Comment:
_imho_ blocks like these have little utility.

Reply:
i suppose it's pretty obvious when looking at the code, removed the blocks.